src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketMirrorHandler.java

```diff
@@ -0,0 +1,129 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactory;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketMirrorClient;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.credentials.JenkinsToBitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepository;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepositoryDescriptor;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketMirroredRepositoryStatus;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import hudson.util.ListBoxModel.Option;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.logging.Level.SEVERE;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+class BitbucketMirrorHandler {
```

Comment:
This feels like a strange class to me. One method returns a `StandardListBoxModel` and the other returns a `BitbucketMirroredRepository`. I feel like these two methods should be in separate classes, or maybe some of the commonality between them can be in this class and the list box stuff can stay out of it. Either way, we're sticking two unrelated concepts into one class which feels wrong

Reply:
I agree it feels a bit strange but so far this has proved useful to me. For example, 

1. if mirror selection does not match the list, the primary one is selected.
2. If there are no mirrors, the primary one is set as selected.
3. Unit tests are really easy to add. Jenkins ListBox is just a List.

Reply:
I feel like `fetchRepositories` is better suited to `BitbucketScmHelper` which is where all the other calls out to Bitbucket live

Reply:
I don't agree. We will lose cohesion. Thinking about it again, BitbucketSearchHelper and BitbucketScmHelper should have been one class which should have handling for the project and repo search boxes.

Reply:
I think I agree with Gaurav here. The classes should be grouped by responsibility rather than what calls they make (arguably quite often these two go hand in hand). This class is a bit weird but overall I think it is mostly ok. 
Having said that I can definitely see it the other way as well, but I have a slight preference for this architecture.